publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -768,32 +765,19 @@ private void setCuratingInstitutions(Resource newResource, CristinUnitsUtil cris
                 newResource.toPublication().getEntityDescription(), cristinUnitsUtil));
     }
 
-    private ImportCandidate insertResourceFromImportCandidate(Resource newResource) {
-        TransactWriteItem[] transactionItems = transactionItemsForNewImportCandidateInsertion(newResource);
-
-        var fileTransactionWriteItems = newResource.getFiles()
-                                            .stream()
-                                            .map(file -> FileEntry.create(file, newResource.getIdentifier(),
-                                                                          UserInstance.fromPublication(
-                                                                              newResource.toPublication())))
-                                            .map(FileEntry::toDao)
-                                            .map(dao -> dao.toPutNewTransactionItem(tableName))
-                                            .toList();
-
+    private ImportCandidate insertResourceFromImportCandidate(ImportCandidateDao importCandidateDao) {
         var transactions = new ArrayList<TransactWriteItem>();
-        transactions.addAll(Arrays.stream(transactionItems).toList());
-        transactions.addAll(fileTransactionWriteItems);
+        var put = new Put()
+                      .withItem(importCandidateDao.toDynamoFormat())
+                      .withTableName(tableName)
+                      .withConditionExpression(KEY_NOT_EXISTS_CONDITION)
+                      .withExpressionAttributeNames(PRIMARY_KEY_EQUALITY_CONDITION_ATTRIBUTE_NAMES);
```

Comment:
Can you annotate the method to explain the relationship between Put and the condition expression